### PR TITLE
Rewrite doc params in Google-style

### PR DESCRIPTION
**Pull request by IlyaFaer**

### CODE REVIEW

File: bigquery/google/cloud/bigquery/_helpers.py [hunk 1]
```diff
@@ -210,11 +210,9 @@ def _row_tuple_from_json(row, schema):
 
     Note:  ``row['f']`` and ``schema`` are presumed to be of the same length.
 
-    :type row: dict
-    :param row: A JSON response row to be converted.
+        row (dict): A JSON response row to be converted.
```
Comment (tswast) [inline bigquery/google/cloud/bigquery/_helpers.py]: We use the names from the [Python typing module](https://docs.python.org/3/library/typing.html) for type definitions. Thus, this should be `Dict`.

File: bigquery/google/cloud/bigquery/_helpers.py [hunk 2]
```diff
@@ -210,11 +210,9 @@ def _row_tuple_from_json(row, schema):
...
 
-    :type schema: tuple
-    :param schema: A tuple of
+        schema (tuple): A tuple of
```
Comment (tswast) [inline bigquery/google/cloud/bigquery/_helpers.py]: If a docstring is multi-line, it should start on the line after the types.

```suggestion
        schema (Tuple[google.cloud.bigquery.schema.SchemaField, ...]):
            A tuple of
```

Also, we should show the type of the values contained in the tuple. See: https://docs.python.org/3/library/typing.html#typing.Tuple

File: bigquery/google/cloud/bigquery/_http.py [hunk 3]
```diff
@@ -22,11 +22,9 @@
 class Connection(_http.JSONConnection):
     """A connection to Google BigQuery via the JSON REST API.
 
-    :type client: :class:`~google.cloud.bigquery.client.Client`
-    :param client: The client that owns the current connection.
+        client (:class:`~google.cloud.bigquery.client.Client`): The client that owns the current connection.
```
Comment (tswast) [inline bigquery/google/cloud/bigquery/_http.py]: No `:class:` for the types in Google-style.

```suggestion
        client (google.cloud.bigquery.client.Client): The client that owns the current connection.
```

File: bigquery/google/cloud/bigquery/client.py [hunk 4]
```diff
@@ -1744,26 +1741,28 @@ def _initiate_resumable_upload(self, stream, metadata, num_retries):
...
+                The "200 OK" response object returned after the multipart
+                upload request.
 
-        :rtype: :class:`~requests.Response`
-        :returns: The "200 OK" response object returned after the multipart
-                  upload request.
-        :raises: :exc:`ValueError` if the ``stream`` has fewer than ``size``
-                 bytes remaining.
+        Raises:
+            :exc:`ValueError` if the ``stream`` has fewer than ``size``
```
Comment (IlyaFaer) [inline bigquery/google/cloud/bigquery/client.py]: This `:exc:` marking should be deleted too? Didn't found anything about it in https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html#example-google

File: bigquery/google/cloud/bigquery/client.py [hunk 5]
```diff
@@ -2348,14 +2347,13 @@ def schema_to_json(self, schema_list, destination):
...
+        iterator (google.api_core.page_iterator.Iterator): The iterator that is currently in use.
 
-    :type resource: dict
-    :param resource: An item to be converted to a project.
+        resource (Dict): An item to be converted to a project.
 
-    :rtype: :class:`.Project`
-    :returns: The next project in the page.
+    Returns:
+        (.Project): The next project in the page.
```
Comment (IlyaFaer) [inline bigquery/google/cloud/bigquery/client.py]: Saw several class names with such periods. Looks abnormal to me

Comment (tswast) [reply bigquery/google/cloud/bigquery/client.py]: I think that refers to the current module. I agree that it looks abnormal. Would probably be better to write this as `google.cloud.bigquery.client.Project` in this case.

File: bigquery/google/cloud/bigquery/client.py (hunk 4 above)
Comment (tswast) [reply bigquery/google/cloud/bigquery/client.py]: Yes, exception types are formatted similarly to return types. I'd suggest the following formatting:

```suggestion
            ValueError:
                If the ``stream`` has fewer than ``size``
```

File: bigquery/google/cloud/bigquery/_helpers.py [hunk 6]
```diff
@@ -210,15 +210,15 @@ def _row_tuple_from_json(row, schema):
...
     Note:  ``row['f']`` and ``schema`` are presumed to be of the same length.
 
-    :type row: dict
-    :param row: A JSON response row to be converted.
+    Args:
+        row (Dict): A JSON response row to be converted.
 
-    :type schema: tuple
-    :param schema: A tuple of
-                   :class:`~google.cloud.bigquery.schema.SchemaField`.
```
Comment (tswast) [inline bigquery/google/cloud/bigquery/_helpers.py]: We'd actually want to retain `:class:` when it appears in the description, just not in the type definition.

File: bigquery/google/cloud/bigquery/_helpers.py [hunk 7]
```diff
@@ -210,15 +210,15 @@ def _row_tuple_from_json(row, schema):
...
+        schema (Tuple):
+            A tuple of
+            google.cloud.bigquery.schema.SchemaField.
 
-    :rtype: tuple
-    :returns: A tuple of data converted to native types.
+    Returns:
+        (Tuple): A tuple of data converted to native types.
```
Comment (tswast) [inline bigquery/google/cloud/bigquery/_helpers.py]: No parentheses around return types.

File: bigquery/google/cloud/bigquery/dataset.py [hunk 8]
```diff
@@ -338,7 +338,7 @@ def _key(self):
         Used to compute this instance's hashcode and evaluate equality.
 
         Returns:
-            Tuple[str]: The contents of this :class:`.DatasetReference`.
+            Tuple[str]: The contents of this .DatasetReference.
```
Comment (tswast) [inline bigquery/google/cloud/bigquery/dataset.py]: This should be reverted, as it's already in Google-style. Descriptions should retain `:class:` annotations.

File: bigquery/google/cloud/bigquery/external_config.py [hunk 9]
```diff
@@ -560,7 +560,7 @@ class ExternalConfig(object):
     """Description of an external data source.
 
     Args:
-        source_format (:class:`~.external_config.ExternalSourceFormat`):
+        source_format (.external_config.ExternalSourceFormat):
```
Comment (tswast) [inline bigquery/google/cloud/bigquery/external_config.py]:
```suggestion
        source_format (google.cloud.bigquery.external_config.ExternalSourceFormat):
```

File: bigquery/google/cloud/bigquery/_helpers.py (hunk 6 above)
Comment (IlyaFaer) [reply bigquery/google/cloud/bigquery/_helpers.py]: Okay, I think I got it. Done

File: bigquery/google/cloud/bigquery/client.py [hunk 10]
```diff
@@ -2426,29 +2423,27 @@ def _item_to_routine(iterator, resource):
...
     return TableListItem(resource)
 
 
 def _make_job_id(job_id, prefix=None):
     """Construct an ID for a new job.
 
-    :type job_id: str or ``NoneType``
-    :param job_id: the user-provided job ID
+    Args:
+        job_id (Union[str, ``NoneType``]): the user-provided job ID
```
Comment (IlyaFaer) [inline bigquery/google/cloud/bigquery/client.py]: Added unioning for some cases

File: bigquery/google/cloud/bigquery/job.py [hunk 11]
```diff
@@ -712,21 +718,21 @@ def done(self, retry=DEFAULT_RETRY):
...
-        :returns: This instance.
+        Returns:
+            _AsyncJob: This instance.
 
-        :raises:
-            :class:`~google.cloud.exceptions.GoogleCloudError` if the job
-            failed or :class:`concurrent.futures.TimeoutError` if the job did
-            not complete in the given timeout.
+        Raises:
+            google.cloud.exceptions.GoogleCloudError, concurrent.futures.TimeoutError:
```
Comment (IlyaFaer) [inline bigquery/google/cloud/bigquery/job.py]: If several types of errors can be raised, should they be written with `Union[]`?

File: bigquery/google/cloud/bigquery/_helpers.py (hunk 7 above)
Comment (IlyaFaer) [reply bigquery/google/cloud/bigquery/_helpers.py]: Done

File: bigquery/google/cloud/bigquery/client.py (hunk 5 above)
Comment (IlyaFaer) [reply bigquery/google/cloud/bigquery/client.py]: Well, in this case yes, but I found few places, in which such statement was pointing to anouther module. Fixed it anyway

File: bigquery/google/cloud/bigquery/client.py (hunk 10 above)
Comment (tswast) [reply bigquery/google/cloud/bigquery/client.py]: `str or NoneType` / `str or None` should transform to `Optional[str]`. https://docs.python.org/3/library/typing.html#typing.Optional

File: bigquery/google/cloud/bigquery/job.py (hunk 11 above)
Comment (tswast) [reply bigquery/google/cloud/bigquery/job.py]: If several types of errors can be raised, they each should appear in their own block. In this example

```
Raises:
    google.cloud.exceptions.GoogleCloudError:
        If the request failed. ...
    concurrent.futures.TimeoutError:
        If the job did not complete in the given timeout.

**Comment by IlyaFaer** [comment]
@tswast, added changes into script
So, I think, everything, that could have been automated is automated. I'll split the changes for BigQuery into two PRs. Also will add some manual changes for anomalous cases and will push all of it shortly
Thank you for cooperating!

**Pull request closed without merging**